src/formatting_stack/strategies.clj

```diff
@@ -80,11 +81,28 @@
        (impl/extract-clj-files)
        (into files)))
 
+(defn current-branch-name []
+  (-> (sh "git" "rev-parse" "--abbrev-ref" "HEAD")
+      (:out)
+      (string/split-lines)
+      (first)))
+
+(defn default-branch-name []
+  (let [all-branches (->> (sh "git" "branch")
+                          :out
+                          string/split-lines
+                          (map (fn [s]
+                                 (-> s (string/split #"\s+") last)))
+                          (set))]
+    (or (some all-branches ["master" "main" "stable" "dev"])
+        ;; return something, for not breaking code that traditionally assumed "master":
+        "master")))
```

Comment:
By returning a known-bad value (master wasn't in `all-branches`, so it's not the right one) the diff-functionality doesn't work iirc.

I think highlighting that mistake / error is preferable to silently doing the wrong thing. With the system-prop suggestion above the error could even suggest setting the system-prop to solve the error.

This could be either an exception; or a simple `log/warn` in the `or`-case. WDYT?

Reply:
We can also observe that no behavior is being changed. I wouldn't be super enthusiastic about breaking it on purpose.

The 'compromise' solution would be to return `"master"` _and_ log a warning (with the sysprop suggestion). sgty?

Reply:
That sounds good 👌🏻